File: Cs-Connect-java/app/src/main/java/com/csestateconnect/www/csconnect/adapters/SearchResultListAdapter.java
package com.csestateconnect.www.csconnect.adapters;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.csestateconnect.www.csconnect.MainActivity;
import com.csestateconnect.www.csconnect.ProjectDetailActivity;
import com.csestateconnect.www.csconnect.ProjectFragment;
import com.csestateconnect.www.csconnect.R;
import com.csestateconnect.www.csconnect.models.search.SearchResult;

import java.util.LinkedList;

public class SearchResultListAdapter extends RecyclerView.Adapter<SearchResultListAdapter.SearchResultViewHolder> {

    private final LinkedList<SearchResult> mSearchResultList;
    private LayoutInflater mInflater;
    private Context mContext;

    public SearchResultListAdapter(Context context, LinkedList<SearchResult> searchResultList) {
        mInflater = LayoutInflater.from(context);
        mContext = context;
        this.mSearchResultList = searchResultList;
    }


    class SearchResultViewHolder extends RecyclerView.ViewHolder {
        public final ImageView logoItemView;
        public final TextView projectName;
        public final TextView projectLocation;
        public final TextView cityName;
        public final LinearLayout searchResultParent;

        final SearchResultListAdapter mAdapter;

        public SearchResultViewHolder(View itemView, SearchResultListAdapter adapter) {
            super(itemView);

            logoItemView = itemView.findViewById(R.id.search_result_image);

            projectName =  itemView.findViewById(R.id.search_result_project_name);
            projectLocation =  itemView.findViewById(R.id.search_result_project_location_name);

            cityName =  itemView.findViewById(R.id.search_result_city_name);

            searchResultParent = itemView.findViewById(R.id.search_result_parent);

            this.mAdapter = adapter;
        }
    }

    @Override
    public int getItemCount() {
        return mSearchResultList.size();
    }

    @Override
    public SearchResultViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View mItemView = mInflater.inflate(R.layout.search_result, parent, false);
        return new SearchResultViewHolder(mItemView, this);
    }

    @Override
    public void onBindViewHolder(SearchResultViewHolder holder, int position) {
        SearchResult mCurrentSearchResult = mSearchResultList.get(position);

        if (mCurrentSearchResult.getType().equals("CITY")){
            holder.projectName.setVisibility(View.GONE);
            holder.projectLocation.setVisibility(View.GONE);

            holder.logoItemView.setImageDrawable(mContext.getResources().getDrawable(R.drawable.ic_location_on_black_24dp));

            holder.cityName.setVisibility(View.VISIBLE);
            holder.cityName.setText(mCurrentSearchResult.getName());

            final String city_name = mCurrentSearchResult.getName();

            holder.searchResultParent.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    Intent intent = new Intent(mContext, MainActivity.class);
                    intent.putExtra("user_city_name", city_name);
                    mContext.startActivity(intent);
                }
            });



        } else {
            holder.projectName.setVisibility(View.VISIBLE);
            holder.projectLocation.setVisibility(View.VISIBLE);

            holder.projectName.setText(mCurrentSearchResult.getName());
            holder.projectLocation.setText(mCurrentSearchResult.getLocation());

            holder.logoItemView.setImageDrawable(mContext.getResources().getDrawable(R.drawable.ic_project));

            holder.cityName.setVisibility(View.GONE);

            final Integer project_id = mCurrentSearchResult.getId();

            holder.searchResultParent.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    Intent intent = new Intent(mContext, ProjectDetailActivity.class);
                    intent.putExtra("project_id", project_id);
                    mContext.startActivity(intent);
                }
            });

        }

    }
}

File: CsConnect/app/src/main/java/com/csestateconnect/adapters/SearchResultListAdapter.kt
package com.csestateconnect.adapters

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.core.os.bundleOf
import androidx.navigation.NavController
import androidx.navigation.Navigation
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.csestateconnect.R
import com.csestateconnect.db.data.SearchResult
import com.csestateconnect.db.data.leads.ListOfLeads
import com.csestateconnect.db.data.projects.*
import kotlinx.android.synthetic.main.cs_assign_lead_items.view.*
import kotlinx.android.synthetic.main.search_result_items.view.*
import java.lang.Exception
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*


class SearchResultListAdapter(val projectResultItems: LinkedList<SearchResult>) :
    RecyclerView.Adapter<SearchResultListAdapter.ViewHolder>() {

    var navController: NavController? = null

    override fun onCreateViewHolder(p0: ViewGroup, viewType: Int): ViewHolder {

        navController = Navigation.findNavController(p0)
        val v: View = LayoutInflater.from(p0.context)
            .inflate(R.layout.search_result_items,p0,false)
        return ViewHolder(v)
    }

    /*class CustomViewHolder(val view:View):RecyclerView.ViewHolder(view)*/
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

        try {
            val mCurrentSearchResult = projectResultItems.get(position)

            if (mCurrentSearchResult.getType().equals("CITY")) {
                holder.projectName.visibility = View.GONE
                holder.projectLocation.visibility = View.GONE
                holder.logoItemView.setImageDrawable(holder.itemView.getResources().getDrawable(R.drawable.ic_location_on_black_24dp))
                holder.cityName.visibility = View.VISIBLE
                holder.cityName.text = mCurrentSearchResult.getName()

                val city_name = mCurrentSearchResult.getName()

                holder.searchResultParent.setOnClickListener {
                    val bundle = bundleOf("city_name" to city_name)
                    navController!!.navigate(R.id.action_searchFragment_to_navigation_projects, bundle)
                }


            } else if(mCurrentSearchResult.getType().equals("PROJECT")){
                holder.projectName.visibility = View.VISIBLE
                holder.projectLocation.visibility = View.VISIBLE

                holder.projectName.text = mCurrentSearchResult.getName()
                holder.projectLocation.text = mCurrentSearchResult.getLocation()
                holder.logoItemView.setImageDrawable(holder.itemView.getResources().getDrawable(R.drawable.ic_project))
                holder.cityName.visibility = View.GONE

                val project_id = mCurrentSearchResult.getId()

                holder.searchResultParent.setOnClickListener {
                    val sharedPreference =  holder.itemView.context.getSharedPreferences("PREFERENCE_NAME", Context.MODE_PRIVATE)
                    val editor = sharedPreference.edit()
                    editor.putString("projectId",project_id.toString())
                    editor.apply()
//                    val bundle = bundleOf("projectId" to project_id)
                    navController!!.navigate(R.id.action_searchFragment_to_projectDetailFragment)
                }

            }
        } catch (e:Exception){
            e.printStackTrace()
        }

    }

    override fun getItemCount(): Int {
        return 1
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView){
        var projectName = itemView.search_result_project_name as TextView
        var projectLocation = itemView.search_result_project_location_name as TextView
        var logoItemView = itemView.search_result_image as ImageView
        var searchResultParent = itemView.search_result_parent as LinearLayout
        var cityName = itemView.search_result_city_name as TextView
    }

//    override fun getItemId(position: Int): Long {
//        return position.toLong()
//    }

//    override fun getItemViewType(position: Int): Int {
//        return position
//    }
}
